.github/workflows/pr-testing-with-test-project.yml

```diff
@@ -39,7 +49,8 @@ jobs:
       - name: Determine if tests should run
         id: should_run
         if: >
-          !(
+          needs.changes.outputs.should_test == 'true' &&
```

Comment:
for now please do not add this condition here but rather add a step above to log the value and the condition result, just, similar to what you did in lines 33-36